lib/client.ex

```diff
@@ -0,0 +1,27 @@
+defmodule RokuIapElixir.Client do
+  use HTTPoison.Base
+
+  def process_url(url) do
+    "https://apipub.roku.com/listen/transaction-service.svc" <> url
+  end
+
+  defp process_response_body(body) do
+    body
+    |> Poison.decode!
+  end
+
+  defp process_request_body(body) do
+    case body do
+      "" -> body
+      _ -> body |> Poison.encode!
+    end
+  end
+
+  defp process_request_headers(headers) do
+    [{"Accept", "application/json"} | headers]
```

Comment:
fwiw, I've had to purposely downcase header keys when overriding. otherwise I end up with "accept" and "Accept"